packages/react/src/components/planning-grid/PlanningGrid.tsx

```diff
@@ -59,12 +62,32 @@ export const PlanningGrid = React.forwardRef<DataGrid, PlanningProps>(({ dataSou
     }
   }, []);
 
-  const navigateToDetails = useCallback(() => {
-    navigate('/planning-task-details');
+  const navigateToDetails = useCallback(({ rowType }: RowClickEvent) => {
+    if (navigation && rowType !== 'detailAdaptive') {
+      navigate('/planning-task-details');
+    }
+  }, []);
+
+  const onEditingStart = useCallback(() => {
+    setNavigation(false);
+  }, []);
+
+  const onEditEnd = useCallback(() => {
```

Comment:
We can merge two the same methods in one, something like: onEditing = () => setNavigation(!navigation);
By the way, the same methods has very different names. They should be called as onEdit**ing**Start/onEdit**ing**End or without **ing** both. This is in all platforms.